overlord/snapstate/snapstate.go

```diff
@@ -1555,8 +1555,16 @@ func RefreshCandidates(st *state.State, user *auth.UserState) ([]*snap.Info, err
 		return nil, err
 	}
 
-	plan, err := refreshCandidates(context.TODO(), st, allSnaps, nil, user, nil, Options{})
-	return plan.targetInfos(), err
+	opts := Options{
+		PrereqTracker: snap.SimplePrereqTracker{},
+	}
+
+	plan, err := refreshCandidates(context.TODO(), st, allSnaps, nil, user, nil, opts)
```

Comment:
I think this was easy to miss because `refreshCandidates` doesn't really just return candidates for refresh anymore. It creates a plan for reconciling the state of snaps with the requested operation so maybe we can change its name? It could clarify what's actually happening here (and why we can't just return infos)

Reply:
Good idea, thank you.